internal/processing/istio/reconciliation_test.go

```diff
@@ -0,0 +1,163 @@
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Reconciliation", func() {
+	When("multiple handlers in addition to Istio JWT", func() {
+		jwtConfigJSON := fmt.Sprintf(`{"authentications": [{"issuer": "%s", "jwksUri": "%s"}]}`, JwtIssuer, JwksUri)
+		jwt := []*gatewayv1beta1.Authenticator{
+			{
+				Handler: &gatewayv1beta1.Handler{
+					Name: "jwt",
+					Config: &runtime.RawExtension{
+						Raw: []byte(jwtConfigJSON),
+					},
+				},
+			},
+		}
+
+		It("with allow handler should provide Istio VS, RA and 2 APs", func() {
+			// given
+			allow := []*gatewayv1beta1.Authenticator{
+				{
+					Handler: &gatewayv1beta1.Handler{
+						Name: "allow",
+					},
+				},
+			}
+
+			allowRule := GetRuleFor(ApiPath, ApiMethods, []*gatewayv1beta1.Mutator{}, allow)
+			jwtRule := GetRuleFor(HeadersApiPath, ApiMethods, []*gatewayv1beta1.Mutator{}, jwt)
+			rules := []gatewayv1beta1.Rule{allowRule, jwtRule}
+
+			apiRule := GetAPIRuleFor(rules)
+			faceClient := GetEmptyFakeClient()
+
+			// when
+			var createdObjects []client.Object
+			reconciliation := istio.NewIstioReconciliation(GetTestConfig())
+			for _, processor := range reconciliation.GetProcessors() {
+				results, err := processor.EvaluateReconciliation(context.TODO(), faceClient, apiRule)
+				Expect(err).To(BeNil())
+				for _, result := range results {
+					createdObjects = append(createdObjects, result.Obj)
+				}
+			}
+
+			// then
+			Expect(createdObjects).To(HaveLen(4))
+
+			vsCreated, raCreated := false, false
+			apCreated := 0
+			for _, createdObj := range createdObjects {
+				vs, vsOk := createdObj.(*networkingv1beta1.VirtualService)
+				ra, raOk := createdObj.(*securityv1beta1.RequestAuthentication)
+				ap, apOk := createdObj.(*securityv1beta1.AuthorizationPolicy)
+
+				if vsOk {
+					vsCreated = true
+					Expect(vs).NotTo(BeNil())
+					Expect(len(vs.Spec.Http)).To(Equal(2))
+					Expect(len(vs.Spec.Http[1].Route)).To(Equal(1))
+					Expect(vs.Spec.Http[1].Route[0].Destination.Host).To(Equal(ServiceName + "." + ApiNamespace + ".svc.cluster.local"))
+				} else if raOk {
+					raCreated = true
+					Expect(ra).NotTo(BeNil())
+					Expect(len(ra.Spec.JwtRules)).To(Equal(1))
+					Expect(ra.Spec.JwtRules[0].Issuer).To(Equal(JwtIssuer))
+					Expect(ra.Spec.JwtRules[0].JwksUri).To(Equal(JwksUri))
+				} else if apOk {
+					apCreated++
+					Expect(ap).NotTo(BeNil())
+					Expect(len(ap.Spec.Rules)).To(Equal(1))
+				}
+			}
+
+			Expect(vsCreated && raCreated && apCreated == 2).To(BeTrue())
+		})
+
+		It("with Ory oauth2 should provide Istio VS, AP, RA and Ory rule", func() {
+			// given
+			oauthConfigJSON := fmt.Sprintf(`{"required_scope": [%s]}`, ToCSVList(ApiScopes))
+			oauth := []*gatewayv1beta1.Authenticator{
+				{
+					Handler: &gatewayv1beta1.Handler{
+						Name: "noop",
```

Comment:
The test description is about oauth2, but the handler user is noop. Also the oauthConfigJSON is used.

Reply:
Yes, this is copy-paste typo.